java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -34,105 +34,110 @@ public class CommandManager {
     /**
      * Build a command and send.
      *
-     * @param command The command to execute
+     * @param requestType Redis command type
+     * @param arguments Redis command arguments
      * @param responseHandler The handler for the response object
      * @return A result promise of type T
      */
     public <T> CompletableFuture<T> submitNewCommand(
-            Command command, RedisExceptionCheckedFunction<Response, T> responseHandler) {
-        // write command request to channel
-        // when complete, convert the response to our expected type T using the given responseHandler
-        return channel
-                .write(
-                        prepareRedisRequest(
-                                command.getRequestType(),
-                                command.getArguments(),
-                                Optional.ofNullable(command.getRoute())),
-                        true)
-                .exceptionally(this::exceptionHandler)
-                .thenApplyAsync(responseHandler::apply);
+            RedisRequestOuterClass.RequestType requestType,
+            String[] arguments,
+            RedisExceptionCheckedFunction<Response, T> responseHandler) {
+
+        RedisRequest.Builder command = prepareRedisRequest(requestType, arguments);
+        return submitCommandToChannel(command, responseHandler);
     }
 
     /**
-     * Exception handler for future pipeline.
+     * Build a command and send.
      *
-     * @param e An exception thrown in the pipeline before
-     * @return Nothing, it rethrows the exception
+     * @param requestType Redis command type
+     * @param arguments Redis command arguments
+     * @param route Command routing parameters
+     * @param responseHandler The handler for the response object
+     * @return A result promise of type T
      */
-    private Response exceptionHandler(Throwable e) {
-        if (e instanceof ClosingException) {
-            channel.close();
-        }
-        if (e instanceof RuntimeException) {
-            // RedisException also goes here
-            throw (RuntimeException) e;
-        }
-        throw new RuntimeException(e);
+    public <T> CompletableFuture<T> submitNewCommand(
+            RequestType requestType,
+            String[] arguments,
+            Route route,
+            RedisExceptionCheckedFunction<Response, T> responseHandler) {
+
+        RedisRequest.Builder command = prepareRedisRequest(requestType, arguments, route);
+        return submitCommandToChannel(command, responseHandler);
+    }
+
+    /**
+     * Take a redis request and send to channel.
+     *
+     * @param command The Redis command request as a builder to execute
+     * @param responseHandler The handler for the response object
+     * @return A result promise of type T
+     */
+    protected <T> CompletableFuture<T> submitCommandToChannel(
```

Comment:
renamed from `submitNewCommand` to `submitCommandToChannel` to be more specific and to being unambiguous with the public methods.